modules/imgproc/src/smooth.dispatch.cpp

```diff
@@ -279,6 +279,24 @@ template<> void getGaussianKernel<ufixedpoint16>(int n, double sigma, int, std::
     }
 }
 
+template<> void getGaussianKernel<ufixedpoint32>(int n, double sigma, int, std::vector<ufixedpoint32>& res)
+{
+    std::vector<softdouble> res_sd;
+    softdouble s0 = getGaussianKernelBitExact(res_sd, n, sigma);
+    CV_UNUSED(s0);
+
+    std::vector<int64_t> fixed_256;
+    softdouble approx_err = getGaussianKernelFixedPoint_ED(fixed_256, res_sd, 16);
```

Comment:
Looks like fractional part bit size is the only actual difference here. I prefer make `fixedShift` constant public and add public typedef with base type to fixedpoint classes and make this template generic.
Probably explicit type cast for could be omitted. In this case public `fixedShift` will be enough.